dev-docs/README.md

```diff
@@ -0,0 +1,19 @@
+# Development Documentation
+
+This directory contains general information useful for people working on the Taskcluster codebase.
+It should be used in concert with the READMEs for the services and libraries, and of course examination of the code itself.
+
+The [Taskcluster monorepo](https://github.com/taskcluster/taskcluster) on Github contains the source code for all of the microservices, a collection of supporting libraries, and more.
+
+## RFCs
```

Comment:
Should we also include a link to the retrospectives repo? I see them as a big part of the story of tc.